File: xtraplatform-api/src/main/java/de/ii/xsf/core/api/Module.java
/**
 * Copyright 2018 interactive instruments GmbH
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 */
package de.ii.xsf.core.api;

/**
 *
 * @author zahnen
 */
public interface Module {

    /**
     *
     * @return the name of the module
     */
    public String getName();

    /**
     *
     * @return the description of the module
     */
    public String getDescription();
    
}

File: xtraplatform-dev/xtraplatform-manager-dev/src/main/resources/manager/js/xsf/devinfo/Module.js
/*
 * Copyright 2017 interactive instruments GmbH
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 */
define([
    "dojo/_base/declare",
    "xsf/api/Module",
    "xsf/devinfo/DevInfoWidget"
],
        function(declare, Module, DevInfoWidget) {
            return declare([Module], {
                getMainMenuItems: function() {
                    return [];
                },
                getTopMenuItems: function() {
                    return [];
                },
                getRightFooterItems: function() {
                    return [
                        {
                            id: "devinfo",
                            widget: DevInfoWidget
                        }
                    ];
                },  
                getSettingsItems: function() {
                    return [];
                },
                getServiceSettingsItems: function() {
                    return [];
                },
                getServiceViews: function() {
                    return [];
                },
                getServiceActions: function() {
                    return [];
                }
            });
        });